### ISSUE TITLE
Probably all versions above `1.6.4` are not working with `preserveSymlinks` tsconfig option

### ISSUE THREAD

**Issue by bayoudhi** [ThomasAribart/json-schema-to-ts#88]
Hey, this is a cool library I really like the idea of inferring TS types from JSON schemas 👌🏻  

Almost all the solutions I found are focused on generating types (via CLIs or code), this library is different & unique.

I found something interesting, all the recent versions (above `1.6.4`) of this library are not working for me, I'm pretty sure my config is correct I'm using Typescript v4 and `strict` mode is enabled.

This is what I [receive](https://user-images.githubusercontent.com/3085156/186945261-623746e6-0237-4bfc-88a1-6431e2a1b574.png), the inferring is not working 😕 Used the [example](https://user-images.githubusercontent.com/3085156/186945227-66134a48-8943-4800-a344-182cf6cc90ee.png) in the Readme.

**The interesting part is [this](https://user-images.githubusercontent.com/3085156/186945774-0a6be513-ff4a-4a62-855a-ef43e13920f1.png)**, it looks like `1.6.4` is the most used version in the last week, so I test it and it's working [perfectly](https://user-images.githubusercontent.com/3085156/186946189-0eb4abc9-14df-4bd5-8f2c-0eda26dffdb6.png)!

**Can anyone please confirm that versions above `1.6.4` work? or at least the recent one `2.5.5`?**  Maybe I have something wrong on my side, thanks.

This is my config:
- node `v16.16.0`
- Typescript `v4.7.4`

**Comment by ThomasAribart** [comment]
Hi @bayoudhi,

Interesting, I tried to reproduce the issue with the same Typescript version without success:

![Capture d’écran 2022-08-28 à 18 10](https://user-images.githubusercontent.com/38014240/187084085-5ff2147d-d592-438b-9709-02ec268cdebd.png)

Can you provide your `tsconfig` as well ?

**Comment by bayoudhi** [comment]
Hi @ThomasAribart

I think I found the cause, it's related to `preserveSymlinks` option in my `tsconfig.json` (as you guessed it's something related to `tsconfig` 🙂)
```ts
// tsconfig.json
{
  "compilerOptions": {
    "preserveSymlinks": true
    // other options
  }
}
```

When turned on, inferring does not work, if I remove it or turn it off (`"preserveSymlinks": false`) inferring works as expected. I used the latest version 2.5.5 btw.

I have a monorepo project with multiple `tsconfig` files and modules, and I'm pretty sure `preserveSymlinks` is required, I need to check that. 

If so can I use this library with my current `tsconfig` file?  Maybe there's something we can do to make it work with `preserveSymlinks`, thanks.
